client/network/sync/src/engine.rs

```diff
@@ -766,9 +1042,7 @@ where
 
 								// Make sure that the newly added block announce validation future
 								// was polled once to be registered in the task.
-								if let Poll::Ready(res) =
-									self.chain_sync.poll_block_announce_validation(cx)
-								{
+								if let Poll::Ready(res) = self.poll_block_announce_validation(cx) {
```

Comment:
While I think the actual polling of block announcements should be in `SyncingEngine`, I think the block announcement code itself should be moved to a separate file. `SyncingEngine` already has quite a bit of stuff and this announcement code is relatively self-contained so it would be better if there was an object like `BlockAnnounceValidator` that would have functions like `register_block_announcement()` for pushing block announcements and then implementing `futures::Stream` for polling those block announcements.

I'd also like to see this particular code about pushing, polling and processing unified under one function if possible.

Reply:
Done